languages/ruby/reference/exercise-concepts/nucleotide-count.md

```diff
@@ -0,0 +1,53 @@
+# Concepts of nucleotide-count
+
+## General
+
+- class level methods: used as the starting point of the exercise. They may be named with `self.[method_name]` notation or with a `self` block. Another way of creating class methods is extending a module.
+- module methods: depending on the approach, they may be used instead of the class methods.
+- factory pattern and object creation: the method `from_dna` may be used to create instances of the main class.
+- instance variables: store the common value/state for the instance.
+- accessors: `att_reader` may be used to access instance variables.
+- access modifiers: `private` may be used to restrict access to the readers.
+- method definition: the `histogram` method needs to be defined.
+- method arguments: the method `from_dna` needs an argument.
+- string: one string is passed as the input.
+- `chars`: may be used to convert the string into an array of characters.
+- strings mutation: `freeze` may be used to prevent string constants from mutating. The directive `frozen_string_literal: true` may also be used for the same purpose.
+- Enumerable module: including `Enumerable` and implementing `each` is one of the possible approaches to this exercise.
```

Comment:
As a concept, this is "mixins" in Ruby.  The details of how to do this may be left to the researcher/coder.

Reply:
Yep, the general concept is "mixins" here. The `Enumerable` module is an implementation of that.

Reply:
Comparable as well, since we have to compare what is given against valid nucleotides.

Reply:
Excellent point!